### ISSUE TITLE
Sometimes 'Infinity' is returned in the converted result 

### ISSUE THREAD

**Issue by vinay-luminar** [pbatey/query-to-mongo#28]
For below example code (example for id 5e8454301455190020332048)

```
var queryToMongo = require("query-to-mongo")
const {criteria, options} = queryToMongo({ test:'5e8454301455190020332048' });
console.log(criteria)
```

The returned result is 
{test: **Infinity**}

For any other id for example for some random id 5ed0a62751dbd6008a0e71ec it works fine. 
Only for few select ids for which example id 5e8454301455190020332048 shown above the result returned has Infinity.

**Comment by pbatey** [comment]
Looks like this is related to the way that values are checked as numbers. That sequence is parsable as a number with [_E notation_](https://en.wikipedia.org/wiki/Scientific_notation) but resolves to a value too large to represent so is returned as **Infinity**.

One solution might be to consider strings that resolve to **Infinity** (or negative **Infinity**) as strings instead of number.

Looks like I could use `Number.isFinite()`

**Comment by pbatey** [comment]
For a workaround, you can wrap the id in quotes to ensure it isn’t interpreted as a Number. Found it in the README:

> - Support for forced string comparison; value in single or double quotes (field='10' or field="10") would force a string compare. Allows for string with embedded comma (field="a,b") and quotes (field="that's all folks").

```
const {criteria, options} = queryToMongo({ test:'"5e8454301455190020332048"' })
```

**Comment by pbatey** [comment]
Fixed in query-to-mongo@0.10.2

Value resolves to a Number only `if (isFinite(value))` - **Infinity** will no longer occur.

**Comment by vinay-luminar** [comment]
thanks for the fix
